cmd/security-secrets-setup/entrypoint.sh

```diff
@@ -24,53 +24,36 @@ set -e
 # thus more graceful termination of all sub-processes if any.
 
 # runtime directory is set per user:
-export XDG_RUNTIME_DIR=${XDG_RUNTIME_DIR:-/run/user/$(echo $(id -u))}
-
-PKI_INIT_RUNTIME_DIR=${XDG_RUNTIME_DIR}/${PKI_INIT_DIR}
+XDG_RUNTIME_DIR=${XDG_RUNTIME_DIR:-/run/user/$(echo $(id -u))}
+PATH="$BASE_DIR:$PATH"
+VAULT_TLS_PATH=${VAULT_TLS_PATH:-/run/edgex/secrets/edgex-vault}
+export XDG_RUNTIME_DIR PATH VAULT_TLS_PATH
 
 # debug output:
 echo XDG_RUNTIME_DIR $XDG_RUNTIME_DIR
-echo PKI_INIT_RUNTIME_DIR $PKI_INIT_RUNTIME_DIR
-
-# configuration for TLS materials:
-PKI_CONFIG_JSON_DIR="res"
-PKI_SETUP_VAULT_FILE=${PKI_CONFIG_JSON_DIR}"/pkisetup-vault.json"
-PKI_SETUP_KONG_FILE=${PKI_CONFIG_JSON_DIR}"/pkisetup-kong.json"
+echo BASE_DIR $BASE_DIR
 
-# check if files exists
-if [ ! -f "${PKI_SETUP_VAULT_FILE}" ]; then
-    echo "Error: certificate config file for Vault is missing"
-    exit 1
+# if running security-secrets-setup subcommand
+# build full command line into positional args
+if [ "$1" = 'generate' -o "$1" = 'cache' -o "$1" = 'import' -o "$1" = 'legacy' ]; then
+    set -- security-secrets-setup "$@"
 fi
 
-if [ ! -f "${PKI_SETUP_KONG_FILE}" ]; then
-    echo "Error: certificate config file for Kong is missing"
-    exit 1
+instvaultscript=""
+posthook=""
+if [ "$1" = 'security-secrets-setup' ]; then
+    # update the start_vault script for vault starting
+    instvaultscript="mv /vault/staging/start_vault.sh /vault/init/start_vault.sh"
+    # grant permissions of folders for vault:vault
+    posthook="chown -R 100:1000 ${VAULT_TLS_PATH}/ /vault/init/"
```

Comment:
```suggestion
    posthook="chown -R 100:1000 ${VAULT_TLS_PATH}"
```

Reply:
`su-exec` doesn't happen until the end of Vault's entrypoint script.  No reason to make /vault/init owned by vault.